### PR TITLE
Show helpful message when both db-kind and a driver are set

### LINKED ISSUE #8291: Quarkus unable to launch when both quarkus.datasource.db-kind and quarkus.datasource.driver are set
I am trying to integrate h2 DB with quarkus. I refereed documentation from the quarkus web site and written a sample application that have one rest controller, which call save operation on h2 database.
But while trying to launch application in dev mode(mvn compile quarkus:dev). I am getting below exception.

```
Failed to start Quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quark
[error.zip](https://github.com/quarkusio/code.quarkus.io/files/4407676/error.zip)

us.arc.deployment.ArcProcessor#registerBeans threw an exception: javax.enterprise.inject.spi.DefinitionException: Producer method io.agroal.api.AgroalDataSource createDataSource_4c2
79770c59fa93dcf179ab15a1363f01f14f595() declares multiple scope type annotations: javax.enterprise.context.ApplicationScoped, javax.enterprise.context.ApplicationScoped
        at io.quarkus.arc.processor.Beans.multipleScopesFound(Beans.java:363)
        at io.quarkus.arc.processor.Beans.createProducerMethod(Beans.java:223)
        at io.quarkus.arc.processor.BeanDeployment.findBeans(BeanDeployment.java:763)
        at io.quarkus.arc.processor.BeanDeployment.registerBeans(BeanDeployment.java:206)
        at io.quarkus.arc.processor.BeanProcessor.registerBeans(BeanProcessor.java:122)
        at io.quarkus.arc.deployment.ArcProcessor.registerBeans(ArcProcessor.java:262)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:938)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:273)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
        at java.lang.Thread.run(Thread.java:748)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:206)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:95)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:45)
        at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:59)
        at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:236)
        at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:39)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:131)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
        at io.quarkus.dev.DevModeMain.start(DevModeMain.java:108)
        at io.quarkus.dev.DevModeMain.main(DevModeMain.java:47)
```

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #8291

### CODE REVIEW

**Comment by geoand** [comment]
@gsmet I completely forgot about this... 
Had I remembered I would have pinged you to perhaps get it in `1.3.2`.

File: extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties [hunk 1]
```diff
@@ -6,7 +6,7 @@ quarkus.datasource.jdbc.max-size=13
 
 quarkus.datasource.users.db-kind=h2
 quarkus.datasource.users.username=username1
-quarkus.datasource.users.jdbc.driver=org.h2.Driver
+quarkus.datasource.users.driver=org.h2.Driver
```
Comment (gsmet) [inline extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties:9]: Why this change, you are creating an invalid configuration for a test that was perfectly valid?

**Review by gsmet — CHANGES_REQUESTED** [review]
Could you have a look at my small comment? This looks odd.

File: extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties (hunk 1 above)
Comment (geoand) [reply extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties:9]: I wanted to show that even with this invalid configuration, things now work. If you try to run the this configuration without the PR, the test will fail with an obscure Arc error.
I can of course duplicate this test if you like to leave the original test intact.

Comment (gsmet) [reply extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties:9]: Yeah, I would prefer to keep the test as is. If you want to test that, please add another test.

Comment (geoand) [reply extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties]: PR updated

**Review by gsmet — APPROVED** [review]
